### PR TITLE
OCPBUGS-19757: bump controller iam policy to v2.4.7

### PR DESCRIPTION
**Pull request by alebedev87**
**Context**
AWS changed the required permissions to create a LB target group: [upstream issue describing the change](https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/2692).

**Why didn't we catch this in CI?**
The problem started to be visible after [AWS CLI commands were published](https://github.com/openshift/aws-load-balancer-operator/pull/97). With these commands the upstream IAM policy started to be used instead of [the generated controller's credentials request](https://github.com/openshift/aws-load-balancer-operator/blob/main/hack/controller/controller-credentials-request.yaml). The difference between the two is that the latter is minified to meet [the AWS limit for inline policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_iam-quotas.html#reference_iam-quotas-entity-length). The minified CredentialsRequest has a broader set of permissions and since it's [used in the CI to create the controller's role](https://github.com/openshift/release/blob/3777337d4c56834709f45de404911575444f744e/ci-operator/step-registry/aws-load-balancer/pre-install-rosa/aws-load-balancer-pre-install-rosa-commands.sh#L53-L54) we never saw the missing permissions.

**Changes**
The main change is a bump of the upstream IAM policy (`assets` directory). However to catch similar problems in the future we need to have the controller's CredentialsRequest to be the same as the upstream's IAM policy. For this we need to stop minifying. This in turn requires significant changes in `iamctl` binary to be able to support:
-  multiple resources from AWS statements because `CredentialsRequest`'s resource field is not a slice (`-s` flag added)
- conditions from AWS statements
- customization of the IAM Policy go function (`-f` flag)

**Result**
The IAM policy is bumped enabling the users of AWS CLI command to use the right set of permissions for the controller's role. The controller's `CredentialsRequest` is not minified anymore so that the role created in ROSA e2e using `ccoctl` is as similar as possible to the upstream.

**Note**
[The IAM policy generated for non-STS clusters](https://github.com/openshift/aws-load-balancer-operator/blob/main/pkg/controllers/awsloadbalancercontroller/iam_policy.go#L10) must still be minified as the AWS limit is still there. A new IAM policy for STS will be added in this PR to be used in case [the STS role is given to the controller](https://github.com/openshift/aws-load-balancer-operator/pull/113).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -127,13 +131,27 @@ vet: ## Run go vet against code.
 
 .PHONY: iamctl-gen
 iamctl-gen: iamctl-build iam-gen
-	$(IAMCTL_BINARY) -i $(IAMCTL_ASSETS_DIR)/iam-policy.json -o $(IAMCTL_OUTPUT_DIR)/$(IAMCTL_OUTPUT_FILE) -p $(IAMCTL_GO_PACKAGE) -c $(IAMCTL_OUTPUT_CR_FILE)
-	go fmt -mod=vendor $(IAMCTL_OUTPUT_DIR)/$(IAMCTL_OUTPUT_FILE)
-	go vet -mod=vendor $(IAMCTL_OUTPUT_DIR)/$(IAMCTL_OUTPUT_FILE)
+	# generate controller's IAM policy without minify.
+	@# This policy is for STS clusters as it's turned into a role policy which is limited to 10240 by AWS.
```
Comment (gcs278) [inline Makefile:135]: Okay, so if I understand this correctly, you *could* minify for STS clusters as well, but by not minifying, you help expose issues with the non-minified IAM Policy, of which QE uses to install clusters and is documented `Option 2. Using the AWS CLI` in `install.md`?

**Comment by gcs278** [comment]
I just had a clarifying question, but thanks for the very detailed description. It took me couple of re-reads, but I think I understand your motiviation.
/approve

File: Makefile (hunk 1 above)
Comment (alebedev87) [reply Makefile:135]: Yes, exactly.

**Pull request merged**